elasticsearch/src/main/java/com/gentics/mesh/search/index/AdminIndexHandler.java

```diff
@@ -93,13 +94,22 @@ public void handleSync(InternalActionContext ac) {
 	}
 
 	public void handleClear(InternalActionContext ac) {
-		searchProvider.clear().andThen(Observable.fromIterable(registry.getHandlers()).flatMapCompletable(handler -> handler.init()))
-			.subscribe(() -> {
-				ac.send(message(ac, "search_admin_index_clear"), OK);
-			}, error -> {
+		db.asyncTx(() -> Single.just(ac.getUser().hasAdminRole())).flatMapCompletable(hasAdminRole -> {
+			if (hasAdminRole) {
+				return searchProvider.clear().andThen(Observable.fromIterable(registry.getHandlers()).flatMapCompletable(handler -> handler.init()));
+			} else {
+				throw error(FORBIDDEN, "error_admin_permission_required");
+			}
+		}).subscribe(() -> {
+			ac.send(message(ac, "search_admin_index_clear"), OK);
+		}, error -> {
+			if (error instanceof GenericRestException) {
+				ac.fail(error);
+			} else {
```

Comment:
Instead of this, create a handler that tests if the user is an admin, kind of like secureAll. Doing this will remove a lot of duplicate code.